ckanext/datagovtheme/tests/test_helpers.py

```diff
@@ -136,3 +136,20 @@ def test_is_tagged_ngda():
 
     assert helpers.is_tagged_ngda(dataset_ngda) is True
     assert helpers.is_tagged_ngda(dataset_non_ngda) is False
+
+##################
+# other data resources
+##################
+
+
+@mock.patch("ckanext.datagovtheme.helpers.get_geoplatform_link")
+def test_is_on_geoplatform(mock_get_geoplatform_link):
```

Comment:
I was trying to figure out an approach for that but didn't end up with something that made sense. I would need to mock a response of `get_geoplatform_link` which could be detected by cypress. how would I do that? cypress just visits the page and checks if an element is in it. the content rendered on the page is managed by ckan. I tried adding an equivalent test in `test_ui_data.py` but ran into a jinja issue about a parent block not existing.

Reply:
Looks like cypress supports intercepting network requests... https://blog.openreplay.com/mocking-and-stubbing-in-cypress-unit-tests/

what say we pair on this and both learn something new?

Reply:
yeah totally!

Reply:
> Looks like cypress supports intercepting network requests... https://blog.openreplay.com/mocking-and-stubbing-in-cypress-unit-tests/
> 
> what say we pair on this and both learn something new?

I think cypress mocking is limited to the frondend XHR traffic. I will be surprised if it can intercept api call made from within ckan python code.

Reply:
going to run the cypress test with the intercept to see if it works. if not i'll remove it.